quartz/components/Head.tsx

```diff
@@ -1,30 +1,172 @@
+    // Initialize options if not set
+    if (!fullOptions) {
+      if (typeof cfg.generateSocialImages !== "boolean") {
+        fullOptions = { ...defaultOptions, ...cfg.generateSocialImages }
+      } else {
+        fullOptions = defaultOptions
+      }
+    }
+
+    // Memoize google fonts
+    if (!fontsPromise) {
+      fontsPromise = getSatoriFont(cfg.theme.typography.header, cfg.theme.typography.body)
+    }
+
+    const slug = fileData.filePath
+    // since "/" is not a valid character in file names, replace with "-"
+    const fileName = slug?.replaceAll("/", "-")
     const title = fileData.frontmatter?.title ?? "Untitled"
-    const description = fileData.description?.trim() ?? "No description provided"
+
+    // Get file description (priority: frontmatter > fileData > default)
+    const fdDescription = fileData.description?.trim()
+    let description = ""
+    if (fdDescription) {
+      description = unescapeHTML(fdDescription)
+    }
+
+    if (fileData.frontmatter?.socialDescription) {
+      description = fileData.frontmatter?.socialDescription as string
+    } else if (fileData.frontmatter?.description) {
+      description = fileData.frontmatter?.description
+    }
+
+    if (cfg.generateSocialImages) {
+      // Generate folders for social images (if they dont exist yet)
+      if (!fs.existsSync(imageDir)) {
+        fs.mkdirSync(imageDir, { recursive: true })
+      }
+
+      if (fileName) {
+        // Generate social image (happens async)
+        generateSocialImage(
+          {
+            title,
+            description,
+            fileName,
+            fileDir: imageDir,
+            fileExt: extension,
+            fontsPromise,
+            cfg,
+          },
+          fullOptions,
+        )
+      }
+    }
+
     const { css, js } = externalResources
 
     const url = new URL(`https://${cfg.baseUrl ?? "example.com"}`)
     const path = url.pathname as FullSlug
     const baseDir = fileData.slug === "404" ? path : pathToRoot(fileData.slug!)
 
     const iconPath = joinSegments(baseDir, "static/icon.png")
-    const ogImagePath = `https://${cfg.baseUrl}/static/og-image.png`
+
+    const ogImageDefaultPath = `https://${cfg.baseUrl}/static/og-image.png`
+    // "static/social-images/filename.ext"
+    const ogImageGeneratedPath = `https://${cfg.baseUrl}/${imageDir.replace(
+      "public/",
+      "",
+    )}/${fileName}.${extension}`
+
+    // Use default og image if filePath doesnt exist (for autogenerated paths with no .md file)
+    const useDefaultOgImage = fileName === undefined || !cfg.generateSocialImages
+
+    // Path to og/social image (priority: frontmatter > generated image (if enabled) > default image)
+    let ogImagePath = useDefaultOgImage ? ogImageDefaultPath : ogImageGeneratedPath
+
+    // TODO: could be improved to support external images in the future
+    // Handle aliases (socialImage, image and cover are supported to ensure obsidian publish support)
```

Comment:
i did a frontmatter refactor so all de-aliasing should happen in `quartz/plugins/transformers/frontmatter.ts`